dotnet-interactive-vscode/src/vscode/notebookProvider.ts

```diff
@@ -0,0 +1,83 @@
+import * as vscode from 'vscode';
+import { ClientMapper } from './../clientMapper';
+import { NotebookFile, InteractiveNotebook } from '../interactiveNotebook';
+
+export class DotNetInteractiveNotebookProvider implements vscode.NotebookProvider {
+    constructor(readonly clientMapper: ClientMapper) {
+    }
+
+    async resolveNotebook(editor: vscode.NotebookEditor): Promise<void> {
+        editor.document.languages = ['dotnet-interactive'];
+
+        let contents = '';
+        try {
+            contents = Buffer.from(await vscode.workspace.fs.readFile(editor.document.uri)).toString('utf-8');
+        } catch {
+        }
+
+        let notebook = InteractiveNotebook.parse(contents);
+        this.clientMapper.addClient(notebook.targetKernelName, editor.document.uri);
+        editor.edit(editBuilder => {
+            for (let cell of notebook.cells) {
+                editBuilder.insert(
+                    0,
+                    cell.content,
+                    cell.language,
+                    cell.kind,
+                    cell.outputs,
+                    {
+                        editable: true,
+                        runnable: true
+                    }
+                );
+            }
+        });
+
+        setTimeout(() => {
+            for (let cell of editor.document.cells) {
+                if (cell.cellKind === vscode.CellKind.Code) {
+                    //
+                }
+            }
+        }, 0);
+    }
+
+    async executeCell(document: vscode.NotebookDocument, cell: vscode.NotebookCell | undefined, token: vscode.CancellationToken): Promise<void> {
+        if (!cell) {
+            // TODO: run everything
+            return;
+        }
+
+        let client = this.clientMapper.getClient(document.uri);
+        if (client === undefined) {
```

Comment:
you ok with null client? if is object just  go c++ style
```
if(client){
  //use it
} else {
  // error handling
}
```

Reply:
This is certainly an error case where we don't know what client (i.e., process) to use to execute the cell.  Any thoughts on an understandable error message to fill the output with in this case?